pkg/logql/ast.go

```diff
@@ -364,6 +365,58 @@ func (e *LineFilterExpr) Filter() (log.Filterer, error) {
 
 	return f, nil
 }
+*/
+func (e *LineFilterExpr) Filter() (log.Filterer, error) {
+
+	acc := make([]log.Filterer, 0)
+	for curr := e; curr != nil; curr = curr.Left {
+
+		switch curr.Op {
+		case OpFilterIP:
+			var err error
+			next, err := log.NewIPLineFilter(curr.Match, curr.Ty)
+			if err != nil {
+				return nil, err
+			}
+			acc = append(acc, next)
+		default:
+			var err error // to avoid `f` being shadowed.
+			next, err := log.NewFilter(curr.Match, curr.Ty)
+			if err != nil {
+				return nil, err
+			}
+			acc = append(acc, next)
+		}
+	}
+
+	// Join all contain filters if there are any.
+	containsAllFilter := log.ContainsAllFilter{}
+	regexpFilters := make()
```

Comment:
What's this doing?

Reply:
I think this whole accumulating logic shouldn't live in the `ast.go` but in the `log` package.

Reply:
This is the next step. I want to move all regex to the end.